internal/engine/live_update_state_tree.go

```diff
@@ -15,27 +15,29 @@ type liveUpdateStateTree struct {
 }
 
 // Create a successful build result if the live update deploys successfully.
-func (t liveUpdateStateTree) createResultSet() store.BuildResultSet {
-	iTargetID := t.iTarget.ID()
-	state := t.iTargetState
-	res := state.LastResult
+func createResultSet(trees []liveUpdateStateTree) store.BuildResultSet {
```

Comment:
it might be easier to leave the existing method as is, and have a method for merging BuildResultSets?

Reply:
i mean, i'd much rather call a single "resultSet" func at the end of BaD execution than call it once for each stateTree in the for loop and keep track of the results to merge later. if you're saying that i should have implemented this function by using the old func as a helper and just merging all the results at the end... yeah you're probably right 🤷‍♀ but feel like it would take as much work now to put back as it would have saved me before. unless you have a _style_ objection to this, i'd just leave it

Reply:
hmmm...ok, though in general we should be trying to converge towards code that's independently testable and reusable. writing lots of bespoke functions like this one means we're going to spend lots of time deleting and rewriting them.